app/models/calendars/academic_calendar.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class AcademicCalendar < ApplicationRecord
+  # relations
+  belongs_to :academic_term
+  belongs_to :calendar_type
+  has_many :calendar_events, dependent: :destroy
+  has_many :unit_calendar_events, dependent: :destroy
+  accepts_nested_attributes_for :calendar_events, allow_destroy: true
+
+  # validations
+  validates :name, presence: true
+  validates :senate_decision_date, presence: true
+  validates :senate_decision_no, presence: true
+  validates :academic_term, uniqueness: { scope: :calendar_type }
+
+  # delegates
+  delegate :name, to: :calendar_type, prefix: :type
```

Comment:
Burda conflict var. Şöyle ki, AcademicCalendar modelinin zaten `name` diye bir alanı var. AcademicCalendar.name dediğimiz de kendisine ait name'i getirecek.

Burada sanırım AcademicCalendar için name gerekli mi, gereksiz mi tereddüte düşmüşsün çünkü migrasyon ve modellerin hep tutarlı iken, burada `name`'e validasyon uyguluyorsun ama migration dosyanda o validasyon yok.

`AcademicCalendar.name` kendi ismini mi dönecek yoksa CalendarType'ın name'ini mi burda bir conflict var. Eğer Type'a ait name'i dönmesi yeterli ise belki de bu modelden name alanını kaldırmak mantıklı olabilir?

Reply:
prefix eklediğim zamanda conflict olmuyor diye biliyorum. Kullanmak istediğim `type_name` idi. db'de validasyonu unutmuşum ekliyorum.

Reply:
hmm, bunu bilmiyordum